cli/cli-improvements.md

```diff
@@ -0,0 +1,330 @@
+* Make resource groups more visible in our documentation, and provide recommendations towards common patterns.
+
+### Non-Goals
+
+* (out of scope) Server-side changes (other than Dashboard).
+
+### User scenarios
+
+* As a Radius beginner, I am utilizing the tutorials to acquaint myself with the minimal necessary concepts for deploying an application using Radius. I prefer not to delve into all the details at this stage.
+
+* I would like to deploy multiple applications to the same environment.
+
+* I would like to deploy a single application to multiple environments.
+
+## User Experience (if applicable)
+
+Most of the CLI commands will change to accodomate the absense of a default resource group. "Impact on CLI commands" section captures the details.
+
+## Design
+
+Today, we establish a default resource group during `rad init` process in `config.yaml`.  All of the `rad` cli commands operate on this resource group, unless a different resource group is specified using option `-g`.
+
+With respect to our user scenarios, this means, 
+
+* A new user has to understand and manage Radius resource groups to be able to deploy a simple tutorial application.
+
+* The user has to pick unique resource names across all applications since they would all be deployed to the same resource group (default behavior).
+
+* The user has to create multiple resource groups and manage deploying the same application to different resource groups in order to have multiple copies of the application.
+
+In order to improve the default experience of `rad` commands, we want to adopt a convention-over-configuration approach in determining the Radius resource group to be operated on. This means,
+
+* we will not establish a default resource group during the 'rad init' process in config.yaml.
+* we will establish the convention that by default, an application ```<app-name>``` to be deployed using environment ```<env-name>``` will be a under the resource-group ```<app-name>-<env-name>```. This resource group will be created if needed during the execution of the command.
+
+The workspace entry created as part of rad init will be as below: 
+
+**Current config.yaml**
+
+```
+workspaces:
+    default: default
+    items:
+        default:
+            connection:
+                context: radius-agent-aks
+                kind: kubernetes
+            environment: /planes/radius/local/resourceGroups/default/providers/Applications.Core/environments/default
+            scope: /planes/radius/local/resourceGroups/default
+```
+
+**Proposed config.yaml**
+
+```
+workspaces:
+    default: default
+    items:
+        default:
+            connection:
+                context: radius-agent-aks
+                kind: kubernetes
+            environment: /planes/radius/local/resourceGroups/default/providers/Applications.Core/environments/default
+```
```

Comment:
so if `scope` is completely dropped from the config, does it mean that instead of deploying to the `default` (aka "scoped") resource group, new resource groups based on environment and app names per the convention will be created each time if no `-g` is specified?